app/models/couch.rb

```diff
@@ -16,6 +16,10 @@ def self.search(params)
       .distinct
   end
 
+  def available_nights
+    self.nights.map { |night| night.date if night.reservation_id.nil? }
```

Comment:
I'm getting an oddly comma-separated array when I try that method @njgheorghita.  And that's pretty much the same thing I get when I do something like `nights.pluck(:date)`.  I think I'm missing something, because  I'm surprised this is working, considering the array of dates I get when I map the nights to their dates.  @njgheorghita, maybe you could show us how this works tomorrow?